src/vm/analysis/type_checker/contexts.rs

```diff
@@ -228,4 +259,20 @@ impl <'a> TypingContext <'a> {
             }
         }
     }
+
+    pub fn add_trait_reference(&mut self, name: &ClarityName, value: &TraitIdentifier) {
+        self.traits_references.insert(name.clone(), TypeSignature::TraitReferenceType(value.clone()));
+    }
+
+    pub fn lookup_trait_reference_type(&self, name: &str) -> Option<&TypeSignature> {
```

Comment:
Should this just return a TraitIdentifier? Any caller can turn it into a TypeSignature if necessary (only would be necessary in `lookup_variable`, and then other callers wouldn't need to destructure it.

Reply:
Fine by me, I was aiming at consistency.
Addressed with https://github.com/blockstack/blockstack-core/pull/1246/commits/1a390cfd526eba06acb788a023a83a953afdcf2c